pageserver/src/pgdatadir_mapping.rs

```diff
@@ -902,6 +902,56 @@ impl<'a, R: Repository> DatadirModification<'a, R> {
         Ok(())
     }
 
+    ///
+    /// Flush changes accumulated so far to the underlying repository.
+    ///
+    /// Usually, changes made in DatadirModification are atomic, but this allows
+    /// you to flush them to the underlying repository before the final `commit`.
+    /// That allows to free up the memory used to hold the pending changes.
+    ///
+    /// Currently only used during bulk import of a data directory. In that
+    /// context, breaking the atomicity is OK. If the import is interrupted, the
+    /// whole import fails and the timeline will be deleted anyway. (XXX: is it
```

Comment:
No, it's not. As specified in https://github.com/neondatabase/neon/pull/1809, broken timeline (b/c of import failure) is just ignored on pageserver startup. 

Related discussion: https://github.com/neondatabase/neon/pull/1764#issuecomment-1134620498